infrastructure/authentication/auth_service_impl.go

```diff
@@ -157,11 +157,33 @@ func getPrioritizedApiUrl(customUrl string, engineUrl string) string {
 		return engineUrl
 	}
 
+	// #IDE-1488 Fix for production debugging issue on the EU server.
+	if isApiSubdomain(defaultUrl, customUrl) {
+		return defaultUrl
+	}
+
 	// Otherwise, return the custom URL set by the user.
 	// FedRAMP and single tenant environments.
 	return customUrl
 }
 
+func isApiSubdomain(domain, subdomain string) bool {
```

Comment:
If I read the code correctly, this method returns, whether the given domain starts with `api.` I do not understand what we want to achieve with the code from L178ff.